From 8f712e0ae30b56cd52166dfb50c2937d2223db1a Mon Sep 17 00:00:00 2001
From: Jan Grulich <grulja@gmail.com>
Date: Tue, 5 Aug 2025 09:52:21 +0200
Subject: [PATCH 3/4] IWYU modules/desktop_capture and modules/video_capture
 for PipeWire

Done using
  find modules -name "*.h" -o -name "*.cc" | xargs tools_webrtc/iwyu/apply-include-cleaner
followed by
  tools_webrtc/gn_check_autofix.py -C out/Default/
and git cl format.

Still keep pipewire/spa headers as ignored until PipeWire with IWYU
annotations is available and until we require version of PipeWire where
we are sure all the headers do exist.

Bug: webrtc:422940461
Change-Id: Idb8a43f77bfe7f5e08b6542211ca1400715093f0
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/401602
Reviewed-by: Philipp Hancke <phancke@meta.com>
Reviewed-by: Ilya Nikolaevskiy <ilnik@webrtc.org>
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Reviewed-by: Alexander Cooper <alcooper@chromium.org>
Commit-Queue: Jan Grulich <grulja@gmail.com>
Cr-Commit-Position: refs/heads/main@{#45289}
---
 modules/video_capture/linux/pipewire_session.cc | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc b/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
index 2c533a0edb..2eb73a47e8 100644
--- a/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc
@@ -10,11 +10,18 @@
 
 #include "modules/video_capture/linux/pipewire_session.h"
 
+#include <pipewire/pipewire.h>
 #include <spa/monitor/device.h>
 #include <spa/param/format-utils.h>
 #include <spa/param/format.h>
+#include <spa/param/param.h>
 #include <spa/param/video/raw.h>
-#include <spa/pod/parser.h>
+#include <spa/pod/iter.h>
+#include <spa/pod/pod.h>
+#include <spa/utils/defs.h>
+#include <spa/utils/dict.h>
+#include <spa/utils/hook.h>
+#include <spa/utils/type.h>
 
 #include <algorithm>
 #include <cstdint>
-- 
2.51.0

